crates/wasmtime/src/config.rs

```diff
@@ -341,6 +343,26 @@ impl Config {
         self
     }
 
+    /// Configures whether or not stacks used for async futures are reset to zero after usage.
+    ///
+    /// When the async_support method is enabled for Wasmtime and the call_async variant of
+    /// calling WebAssembly is used then Wasmtime will create a separate runtime execution
+    /// stack for each future produced by call_async. When using the pooling instance
+    /// allocator this allocation will happen from a pool of stacks and additionally
+    /// deallocation will simply release the stack back to the pool. During the deallocation
+    /// process Wasmtime will by default reset the contents of the stack back to zero.
```

Comment:
This last sentence will need an update since this PR is also changing the defaults.